tests/test_pdf.py

```diff
@@ -254,6 +254,17 @@ def test_with_block_abuse(resources):
         im0.read_bytes()
 
 
+def test_allow_overwriting_input(resources):
+    with pikepdf.open(resources / 'pal-1bit-trivial.pdf',
+                      allow_overwriting_input=True) as pdf:
+        with pdf.open_metadata() as meta:
+            meta['dc:title'] = 'New Title'
+        pdf.save()
```

Comment:
Do something like

```python
pdf.save('other.pdf', encryption=dict(owner="owner"))
pdf.save()
pdf.save(linearize=True)
```

to confirm that we can save multiple times.

Reply:
Good idea! Added